### PR TITLE
added two functions requested by wet lab crew

### PR DESCRIPTION
**Pull request by qiyunzhu**
Upon the request from the wet lab crew I added two small functions:

1) To be consistent with QIIME/Qiita standard, underscores and dashes in the sample IDs are replaced by dots.

2) Empty lines (wells) are removed from the output mapping files. This is optional (controlled by the new --empty argument) but is made default.

### CODE REVIEW

File: plate_mapper/plate_mapper.py [hunk 1]
```diff
@@ -54,8 +55,9 @@ def plate_mapper(input_f, barseq_f, output_f, names_f=None, special_f=None):
         Sample name list file
     special_f : file object (optional)
         Special sample definition file
+    empty : bool
```
Comment (josenavas) [inline plate_mapper/plate_mapper.py]: add `, optional` and add into the description that the default value is false.

**Review by josenavas — APPROVED** [review]
Just a minor comment

**Review by mortonjt — APPROVED** [review]
Looks good.  Will merge once corrections are made

File: plate_mapper/plate_mapper.py (hunk 1 above)
Comment (qiyunzhu) [reply plate_mapper/plate_mapper.py]: Thank you! Edited.

**Comment by qiyunzhu** [comment]
Hi @josenavas and @mortonjt , thank you for the review!

File: plate_mapper/plate_mapper.py [hunk 2]
```diff
@@ -54,8 +55,9 @@ def plate_mapper(input_f, barseq_f, output_f, names_f=None, special_f=None):
         Sample name list file
     special_f : file object (optional)
         Special sample definition file
+    empty : bool (optional)
```
Comment (sjanssen2) [inline plate_mapper/plate_mapper.py:58]: how do you define empty? I got a mapping file from Greg containing lines that are composed of a lot of \t but no real characters.

File: plate_mapper/plate_mapper.py [hunk 3]
```diff
@@ -153,8 +155,12 @@ def plate_mapper(input_f, barseq_f, output_f, names_f=None, special_f=None):
                 # empty well (if defined as a special sample)
                 metadatum = specs['']['metadatum']
                 sample = '%s%s.%s' % (specs['']['name'], pid, well)
-        output_f.write('%s\t%s\n' % ('\t'.join((sample, barcode, primer, plate,
-                                     well)), '\t'.join(metadatum)))
+        if sample or empty:
+            # replace underscore and dash with dot in sample name
+            sample_dot = sample.replace('_', '.').replace('-', '.')
+            output_f.write('%s\t%s\n' % (
+                '\t'.join((sample_dot, barcode, primer, plate, well)),
```
Comment (sjanssen2) [inline plate_mapper/plate_mapper.py:162]: why do you have two join statements. Can't you move 'metadatum' into the list with sample_dot, barcode, ...?

File: plate_mapper/plate_mapper.py [hunk 4]
```diff
@@ -200,7 +206,7 @@ def plate_mapper(input_f, barseq_f, output_f, names_f=None, special_f=None):
 if __name__ == "__main__":
     # Welcome information
     print('Plate Mapper: Convert a plate map file into a mapping file.\n'
-          'Last updated: Feb 7, 2017.')
+          'Last updated: Jun 22, 2017.')
```
Comment (sjanssen2) [inline plate_mapper/plate_mapper.py:209]: I like that!

File: plate_mapper/plate_mapper.py (hunk 2 above)
Comment (qiyunzhu) [reply plate_mapper/plate_mapper.py:58]: Here "empty" is a well without sample. The test data files in the repository demonstrates empty-allowed vs empty-free mapping files.

File: plate_mapper/plate_mapper.py (hunk 4 above)
Comment (qiyunzhu) [reply plate_mapper/plate_mapper.py:209]: Thanks!

File: plate_mapper/plate_mapper.py (hunk 3 above)
Comment (qiyunzhu) [reply plate_mapper/plate_mapper.py:162]: I don't think that will work. The former `join` has two sets of parentheses, meaning that `sample_dot`, `barcode`, etc. are scalars. But the latter `join` directly operates on a list (`metadatum`).

**Pull request merged**